clients/src/main/java/org/apache/kafka/common/errors/AbortableTransactionException.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.common.errors;
+
+public class AbortableTransactionException extends ApiException {
```

Comment:
I know there were some questions about the naming -- this seems reasonable but just wanted to call it out.

cc: @artemlivshits

Reply:
I'm still thinking on this name as well. Since all transactions are technically abortable, and what we want to convey is that the only way forward is to abort, would it make sense to say TransactionAbortableException?

Maybe it doesn't make a huge difference though. 😅 Wondering if there is a way to make that distinction of abortable vs we need to abort to proceed. I realize though that we are also modeling off of RetriableException here and trying to keep that similar.

Reply:
I tend to agree that `TransactionAbortableException` is very slightly more satisfying than `AbortableTransactionException`.